internal/packager/bundle.go

```diff
@@ -0,0 +1,101 @@
+package packager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/app/internal/store"
+	"github.com/docker/app/types"
+	"github.com/docker/app/types/metadata"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/distribution/reference"
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func MakeBundleFromApp(dockerCli command.Cli, app *types.App, refOverride reference.NamedTagged) (*bundle.Bundle, error) {
+	logrus.Debug("Making app bundle")
+	meta := app.Metadata()
+	invocationImageName, err := MakeInvocationImageName(meta, refOverride)
+	if err != nil {
+		return nil, err
+	}
+
+	buildContext := bytes.NewBuffer(nil)
+	if err := PackInvocationImageContext(dockerCli, app, buildContext); err != nil {
+		return nil, err
+	}
+
+	logrus.Debugf("Building invocation image %s", invocationImageName)
+	buildResp, err := dockerCli.Client().ImageBuild(context.TODO(), buildContext, dockertypes.ImageBuildOptions{
```

Comment:
Do we still need to build the invocation image? I thought it was handled by buildx? 🤔

Reply:
Yes, I left this one behind as a few other commands (render, install) are tested based on dockerapp folder, didn't want to change everything at once.